portal.go

```diff
@@ -832,6 +900,11 @@ func (portal *Portal) HandleMatrixMessage(evt *event.Event) {
 		}
 	}
 
+	if time.Now().Sub(time.Unix(evt.Timestamp, 0)).Seconds() > 10 {
+		portal.log.Debug("It's been over 10 seconds since the message arrived at the homeserver. Will not handle the event.")
+		return
+	}
```

Comment:
This looks like it's just dropping the event silently? It at least needs to send the correct failure checkpoint, and maybe also an error message. Also not sure if it's a good idea to force a 10 second limit for everyone, might need to make it configurable.